src/cpp/session/modules/SessionSource.cpp

```diff
@@ -289,6 +291,18 @@ Error saveDocumentCore(const std::string& contents,
       pDoc->setFolds(jsonFoldSpec.get_str());
    }
 
+   bool hasMarksSpec = json::isType<std::string>(jsonMarksSpec);
+   if (hasMarksSpec)
+   {
+      pDoc->setMarks(jsonMarksSpec.get_str());
+   }
+
+   bool hasSelectionSpec = json::isType<std::string>(jsonSelectionSpec);
+   if (hasSelectionSpec)
+   {
+      pDoc->setSelection(jsonSelectionSpec.get_str());
```

Comment:
Can we remove the code that sets and restores the `SourcePosition` when popping out/in now that it's done here?